### PR TITLE
Create LinkApp

### PR DESCRIPTION
**Pull request by vilsbole**
Common component for Links.
Toggles between anchor tag for external links and react-router-dom `Link` for history aware anchors.
- Affects Dashboard Top right navlinks
- Busy Bottoms
- TokenList

[[ch7877]](https://app.clubhouse.io/mycrypto/story/7877/link-fixes-previews-and-right-clicking)

### CODE REVIEW

**Review by blurpesec — COMMENTED** [review]
Is this supposed to be a shared Link component? Because we have like a bajillion of those throughout the app. What specific use case is this supposed to address that none of the others do? Can any of them be consolidated to include this functionality?

Some of the ones I found:
- [ExternalLink](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/features/Layout/Navigation/components/ExternalLink.tsx)
- [Link](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/Link.tsx)
- [LinkOut](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/LinkOut.tsx)
- [NewTabLink](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/NewTabLink.tsx)
- [RouterLink](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/RouterLink.tsx)
- [HelpLink](https://github.com/MyCryptoHQ/MyCrypto/blob/master/src/components/HelpLink.tsx)

**Comment by vilsbole** [comment]
Yes this is the shared link component.
The one to unite all others. 
It's destined to replace all others.

File: src/components/LinkApp.test.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  // https://security.stackexchange.com/a/241570
+  it('sets noreferrer attribute on tag', () => {
+    const props = {
+      href: 'https://help.example.com',
+      isExternal: true,
+      children: 'Get help'
+    } as Props;
+    getComponent(props);
+    expect(screen.getByText(`${props.children}`).closest('a')).toHaveAttribute('rel', 'noreferrer');
+  });
```
Comment (GuillaumeRx) [inline src/components/LinkApp.test.tsx:53]: 😍

File: src/components/LinkApp.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        target="_blank"
+        rel="noreferrer"
+        onClick={handleClick}
+        {...props}
+      />
+    );
+  } else {
+    return <SRouterLink to={href} variant={variant} onClick={handleClick} {...props} />;
+  }
+};
```
Comment (GuillaumeRx) [inline src/components/LinkApp.tsx:84]: Using a ternary can save the use a return statement and improve readability IMO

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: src/components/LinkApp.tsx (hunk 2 above)
Comment (vilsbole) [reply src/components/LinkApp.tsx:84]: Done

**Review by GuillaumeRx — APPROVED** [review]
Lgtm!

**Pull request merged**